pkgs/development/python-modules/sounddevice/default.nix

```diff
@@ -8,20 +8,23 @@
 
 buildPythonPackage rec {
   pname = "sounddevice";
-  version = "0.3.9";
+  version = "0.3.11";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1c9e833f8c8ccc67c0291c3448b29e9acc548fe56d15ee6f7fdd7037e00319f8";
+    sha256 = "0pfcbgbl77nggchxb2i5rb78m7hpgn65aqpz99yfx1fgfbmy9yg1";
   };
 
   propagatedBuildInputs = [ cffi numpy portaudio ];
 
   # No tests included nor upstream available.
   doCheck = false;
 
-  prePatch = ''
-    substituteInPlace src/sounddevice.py --replace "'portaudio'" "'${portaudio}/lib/libportaudio.so.2'"
+  patches = [ ./fix-portaudio-library-path.patch ];
```

Comment:
You should use `substituteAll` here instead of an additional `postPatch` phase. Have a look at e.g. https://github.com/NixOS/nixpkgs/blob/a44d33aac12a24914cb35740fd8f256f2ee9c1bc/pkgs/applications/networking/syncthing-gtk/default.nix#L34-L44

Reply:
Thanks for your feedback. I've updated the patch to use substituteAll.